From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tariq Bashir <tbashir@brightsign.biz>
Date: Wed, 28 Jun 2023 16:37:03 +0100
Subject: OS-14354 BrightSign: add remote-debugging-address to content switches

A change has been added to Electron TCPServerSocketFactory to allow a custom
address to be used for remote debugging (i.e. use 0.0.0.0 instead of 127.0.0.1).
This change adds the remote-debugging-address switch to the content switches to
allow it to be used in Electron.

diff --git a/content/public/common/content_switches.cc b/content/public/common/content_switches.cc
index 5e6312932d7d2406a4743f5c7a1743ea308993c4..0b790d5e7d596bb86962a190083c8f0bb48d37b7 100644
--- a/content/public/common/content_switches.cc
+++ b/content/public/common/content_switches.cc
@@ -605,6 +605,12 @@ const char kReduceUserAgentMinorVersion[] = "reduce-user-agent-minor-version";
 // "JSON" (the default) or "CBOR".
 const char kRemoteDebuggingPipe[] = "remote-debugging-pipe";
 
+// Use the given address instead of the default loopback for accepting remote
+// debugging connections. Should be used together with --remote-debugging-port.
+// Note that the remote debugging protocol does not perform any authentication,
+// so exposing it too widely can be a security risk.
+const char kRemoteDebuggingAddress[] = "remote-debugging-address";
+
 // Enables remote debug over HTTP on the specified port.
 const char kRemoteDebuggingPort[]           = "remote-debugging-port";
 
diff --git a/content/public/common/content_switches.h b/content/public/common/content_switches.h
index 1a44e07a2d428f0356ec5285557b3be942269a21..1beca6ead4f2c8f09b13aac73d4dcecbb275b90d 100644
--- a/content/public/common/content_switches.h
+++ b/content/public/common/content_switches.h
@@ -172,6 +172,7 @@ CONTENT_EXPORT extern const char kReduceAcceptLanguageHTTP[];
 CONTENT_EXPORT extern const char kReduceUserAgentMinorVersion[];
 CONTENT_EXPORT extern const char kRemoteDebuggingPipe[];
 CONTENT_EXPORT extern const char kRemoteDebuggingPort[];
+CONTENT_EXPORT extern const char kRemoteDebuggingAddress[];
 CONTENT_EXPORT extern const char kRemoteAllowOrigins[];
 CONTENT_EXPORT extern const char kRendererClientId[];
 extern const char kRendererCmdPrefix[];
